### PR TITLE
Add kotlin IR, update gradle, add spotless to modules

### LINKED ISSUE #2355: Should we use the new JVM IR backend?
Read more [here](https://blog.jetbrains.com/kotlin/2021/02/the-jvm-backend-is-in-beta-let-s-make-it-stable-together/).

### PR DESCRIPTION
**Pull request by EmmanuelMess**
## Description

* Add kotlin IR
* Update gradle
* Add spotless to modules

#### Issue tracker   

Fixes #2355

#### Automatic tests

- [ ] Added test cases
  
#### Manual tests
- [ ] Done  
  

#### Build tasks success  

Successfully running following tasks on local:
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

### CODE REVIEW

**Review by VishnuSanal — APPROVED** [review]

**Review by VishalNehra — APPROVED** [review]

**Comment by EmmanuelMess** [comment]
@VishalNehra wait, did codacy break?

**Comment by EmmanuelMess** [comment]
I'll retrigger the workflow.

**Comment by EmmanuelMess** [comment]
Nvm, I can't trigger codacy.

File: app/build.gradle [hunk 1]
```diff
@@ -95,6 +95,7 @@ android {
 
     kotlinOptions {
         jvmTarget = '1.8'
+        useIR = true
```
Comment (VishalNehra) [inline app/build.gradle:98]: I missed this change, what's the use of this?

Comment (EmmanuelMess) [reply app/build.gradle:98]: Here is the explanation: https://blog.jetbrains.com/kotlin/2021/02/the-jvm-backend-is-in-beta-let-s-make-it-stable-together/

**Pull request merged**